test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -0,0 +1,1083 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2016 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import random
+import string
+import urllib2
+from tests.functional import *
+
+
+class TestCgroupsHook(TestFunctional):
+    """
+    This test suite targets Linux Cgroups hook functionality.
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        self.server.expect(SERVER, {'pbs_version': (GE, '14.0')},
+                           max_attempts=3)
+        self.paths = self.get_paths()
+        if not self.paths:
+            self.skipTest("No cgroups mounted")
+        self.server.set_op_mode(PTL_CLI)
+        self.server.cleanup_jobs(extend='force')
```

Comment:
Just making sure we're starting with a clean slate, I guess. If you're certain I should remove it, I will do so.